l1-contracts/test/foundry/unit/concrete/state-transition/chain-deps/facets/Mailbox/RequestL2Transaction.t.sol

```diff
@@ -0,0 +1,130 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.24;
+
+import {MailboxTest} from "./_Mailbox_Shared.t.sol";
+import {BridgehubL2TransactionRequest} from "contracts/common/Messaging.sol";
+import {REQUIRED_L2_GAS_PRICE_PER_PUBDATA, MAX_NEW_FACTORY_DEPS, ETH_TOKEN_ADDRESS} from "contracts/common/Config.sol";
+import {TransactionFiltererTrue} from "contracts/dev-contracts/test/DummyTransactionFiltererTrue.sol";
+import {TransactionFiltererFalse} from "contracts/dev-contracts/test/DummyTransactionFiltererFalse.sol";
+import {FeeParams, PubdataPricingMode} from "contracts/state-transition/chain-deps/ZkSyncHyperchainStorage.sol";
+import {IL1SharedBridge} from "contracts/bridge/interfaces/IL1SharedBridge.sol";
+
+contract MailboxequestL2TransactionTest is MailboxTest {
+    address tempAddress;
+    bytes[] tempBytesArr;
+    bytes tempBytes;
+
+    function setUp() public virtual {
+        prepare();
+
+        tempAddress = makeAddr("temp");
+        tempBytesArr = new bytes[](0);
+        tempBytes = "";
+        utilsFacet.util_setChainId(eraChainId);
+    }
+
+    function test_RevertWhen_NotEra() public {
+        utilsFacet.util_setChainId(eraChainId + 1);
```

Comment:
Fuzz chainId. Do not hard-code random values. 

Example:

Reply:
fixed